src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,179 @@
+
+  import InputField from "./InputField.svelte";
+  import Button from "./Button.svelte";
+
+  var spreadsheetId = "";
+  var error = null;
+
+  // Client ID and API key from the Developer Console
+  var CLIENT_ID = process.env.G_CLIENT_ID;
+  var API_KEY = process.env.G_API_KEY;
+
+  // Array of API discovery doc URLs for APIs used by the quickstart
+  var DISCOVERY_DOCS = [
+    "https://sheets.googleapis.com/$discovery/rest?version=v4",
+  ];
+
+  // Authorization scopes required by the API; multiple scopes can be
+  // included, separated by spaces.
+  var SCOPES = "https://www.googleapis.com/auth/spreadsheets.readonly";
+
+  // let authorizeButton = document.getElementById("authorize_button");
+  // let signoutButton = document.getElementById("signout_button");
+
+  /**
+   *  On load, called to load the auth2 library and API client library.
+   */
+  function handleClientLoad() {
+    gapi.load("client:auth2", initClient);
+  }
+
+  /**
+   *  Initializes the API client library and sets up sign-in state
+   *  listeners.
+   */
+  function initClient() {
+    error = null;
+    gapi.client
+      .init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES,
+      })
+      .then(
+        function () {
+          // Listen for sign-in state changes.
+          gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+          // Handle the initial sign-in state.
+          updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        },
+        function (error) {
+          error = JSON.stringify(error, null, 2);
+        }
+      );
+  }
+
+  /**
+   *  Called when the signed in status changes, to update the UI
+   *  appropriately. After a sign-in, the API is called.
+   */
+  function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+      getValuesFromSpreadSheet();
+    } else {
+      handleAuthClick();
+      getValuesFromSpreadSheet();
+    }
+  }
+
+  /**
+   *  Sign in the user upon button click.
+   */
+  function handleAuthClick(event) {
+    gapi.auth2.getAuthInstance().signIn();
+  }
+
+  /**
+   *  Sign out the user upon button click.
+   */
+  function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+  }
+
+  /**
+   * Gets the unique spreadsheet ID from a Google Sheet URL
+   *
+   */
+  function getSpreadsheetId() {
+    let url = document.getElementById("googleSheetsURL").value;
+    if (url) {
+      return url.split("/")[5];
+    } else {
+      error =
+        "Error: The url " +
+        url +
+        " is not a valid Google Sheets URl, please paste a valid one.";
+    }
+  }
+
+  /**
+   * Read word and count form storage
+   * Sample: https://docs.google.com/spreadsheets/d/1lzHxoMWHpdGecby4d0y15AVf80csA7iNkaMMEL54Q7g/edit#gid=0
+   */
+  async function getValuesFromSpreadSheet() {
+    const wordListObject = [];
+    spreadsheetId = getSpreadsheetId() || "";
+    if (!error) {
```

Comment:
Suggest refactoring this to use a guard clause, so that the "meat" of this function becomes unindented by one level. It's ever so slightly cleaner!

Here are a few tutorials:
 - https://www.catalyst.net.nz/blog/early-returnsguard-clauses-javascript-and-react
 - https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/